compose/foundation/foundation/src/linuxMain/kotlin/androidx/compose/foundation/internal/ClipboardUtils.linux.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:OptIn(ExperimentalComposeUiApi::class)
+
+package androidx.compose.foundation.internal
+
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.platform.ClipEntry
+import androidx.compose.ui.text.AnnotatedString
+
+
+internal actual suspend fun ClipEntry.readText(): String? {
+    return null
```

Comment:
Many features in this PR are not implemented (this is just one random example). I am afraid that using wayland code here could break it for non-wayland apps.

My use case is using Compose on an embedded system which is not using wayland but manufacturer specific APIs for most (windowing) features. I need to implement some of these functions with manufacturer-specific code, which of course can never be merged into compose.

As this is an expect/actual (like almost everywhere), I do not know of an easy way to pull out the wayland code into a separate module. Ideally a user like me can provide their own implementation for the windowing features. Or maybe multiple separate modules (one for wayland, etc). Any ideas what would work best?